Sources/NIOPosix/MultiThreadedEventLoopGroup.swift

```diff
@@ -138,29 +140,73 @@ public final class MultiThreadedEventLoopGroup: EventLoopGroup {
     /// - arguments:
     ///     - numberOfThreads: The number of `Threads` to use.
     public convenience init(numberOfThreads: Int) {
-        self.init(numberOfThreads: numberOfThreads, selectorFactory: NIOPosix.Selector<NIORegistration>.init)
+        self.init(numberOfThreads: numberOfThreads,
+                  canBeShutDown: true,
+                  selectorFactory: NIOPosix.Selector<NIORegistration>.init)
+    }
+
+    /// Create a ``MultiThreadedEventLoopGroup`` that cannot be shut down and must not be `deinit`ed.
+    ///
+    /// This is only useful for global singletons.
+    public static func _makePerpetualGroup(threadNamePrefix: String,
```

Comment:
Does this need to be `public` now that we have it in the same module?

Reply:
It doesn't _need_ to be but I wanted to keep it as "SPI" to allow other large projects to make their customised singletons too if they already have an existing, similar solution.

Reply:
Will this _actually_ be used? Or is it maybe we'd use this in the future? If it's the former then we should just make it API (should it be "unsafe" too?). If it's the latter then let's keep it internal until it's needed.

Reply:
Only (minor) concern was that if it's public API and you hold it wrong then you can hit an assertion failure. 

If it remains underscored then I don't mind at all. Presumably you're fine with the caveats of it not being public API.

Reply:
fine by me!

Reply:
`@_spi(WD40)` scuse me just passing by.